Meta/Lagom/Tools/CodeGenerators/LibWeb/WrapperGenerator.cpp

```diff
@@ -1275,6 +1345,49 @@ static void generate_arguments(SourceGenerator& generator, Vector<IDL::Parameter
     arguments_builder.join(", ", parameter_names);
 }
 
+// https://webidl.spec.whatwg.org/#create-sequence-from-iterable
+void IDL::ParameterizedType::generate_sequence_from_iterable(SourceGenerator& generator, String const& cpp_name, String const& iterable_cpp_name, String const& iterator_method_cpp_name, HashMap<String, IDL::Dictionary> const& dictionaries, size_t recursion_depth) const
+{
+    auto sequence_generator = generator.fork();
+    sequence_generator.set("cpp_name", cpp_name);
+    sequence_generator.set("iterable_cpp_name", iterable_cpp_name);
+    sequence_generator.set("iterator_method_cpp_name", iterator_method_cpp_name);
+    sequence_generator.set("recursion_depth", String::number(recursion_depth));
+    sequence_generator.set("sequence.type", idl_type_name_to_cpp_type_name(parameters.first()));
+
+    // To create an IDL value of type sequence<T> given an iterable iterable and an iterator getter method, perform the following steps:
+    // 1. Let iter be ? GetIterator(iterable, sync, method).
+    // 2. Initialize i to be 0.
+    // 3. Repeat
+    //      1. Let next be ? IteratorStep(iter).
+    //      2. If next is false, then return an IDL sequence value of type sequence<T> of length i, where the value of the element at index j is Sj.
+    //      3. Let nextItem be ? IteratorValue(next).
+    //      4. Initialize Si to the result of converting nextItem to an IDL value of type T.
+    //      5. Set i to i + 1.
+
+    sequence_generator.append(R"~~~(
+    auto iterator@recursion_depth@ = TRY(JS::get_iterator(global_object, @iterable_cpp_name@, JS::IteratorHint::Sync, @iterator_method_cpp_name@));
+
+    Vector<@sequence.type@> @cpp_name@;
```

Comment:
Hmm, can `@sequence.type@` ever be a GC-allocated type here? If so we need to use a `MarkedValueList`

Reply:
Good point, yes it can, for example if you use `sequence<any>`, `sequence<object>`, `sequence<ArrayBuffer>`, etc.
I've made it so `idl_type_name_to_cpp_type_name` (now named `idl_type_name_to_cpp_type`) returns what storage type the given type can be safely stored in.